### PR TITLE
Migrated Organization app to CBV

### PR DESCRIPTION
**Pull request by amruthasangeeth**
Migrated organization app from function based views to class based views. The following are the views used:
ListView - To list the orrganization
UpdateView - To edit the Oganizationdetails
DeleteView - To delete a Organization object

### CODE REVIEW

**Comment by tapasweni-pathak** [comment]
cc @nida @smarshy.

Great if you both can have a look it it.

**Comment by smarshy** [comment]
@amruthasangeeth This has fewer errors, might be easier to fix. Just check if organization has volunteers, if it doesn't then delete else give error.
I will pull these but can you please check this on your local once?

File: vms/organization/templates/organization/delete.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 {% block setting_content %}
     <div class="spacer"></div>
-    <form class="form-horizontal" action="{% url 'organization:delete' organization_id %}" method="post">
+    <form class="form-horizontal" action="" method="post">
```
Comment (smarshy) [inline vms/organization/templates/organization/delete.html]: @amruthasangeeth Is this redirecting to any view?

Comment (amruthasangeeth) [inline vms/organization/templates/organization/delete.html]: @smarshy Yes it redirects to OrganizationDeleteView

**Comment by smarshy** [comment]
@amruthasangeeth  It is failing on this line - [link](https://github.com/systers/vms/blob/develop/vms/administrator/tests/test_settings.py#L1790)

That is because it cannot detect the error box.
Can you please check this locally once? Create a organization, register a volunteer to it and then try deleting the org from the ui. Check if you get an error message.

I think there is a missing template issue here. On deletion error it directs to delete error template given here -[template](https://github.com/systers/vms/blob/develop/vms/organization/templates/organization/delete_error.html) . Have you redirected it here on delete error?

**Comment by amruthasangeeth** [comment]
@tapasweni-pathak @nida This is also fixed.

File: vms/organization/templates/organization/delete.html (hunk 1 above)
Comment (tapasweni-pathak) [inline vms/organization/templates/organization/delete.html]: @amruthasangeeth Where is it specified that it is redirecting to OrganizationDeleteView?

File: vms/organization/views.py [hunk 2]
```diff
@@ -2,105 +2,80 @@
...
-        admin = user.administrator
-    except ObjectDoesNotExist:
-        pass
-
-    # check that an admin is logged in
-    if admin is not None:
-        return True
-    else:
-        return False
-
```
Comment (tapasweni-pathak) [inline vms/organization/views.py]: @amruthasangeeth Did you delete this check,  'admin is logged in or not'?

File: vms/organization/views.py [hunk 3]
```diff
@@ -2,105 +2,80 @@
...
-                return HttpResponseRedirect(reverse('organization:list'))
-            else:
-                return render(
-                    request,
-                    'organization/create.html',
-                    {'form': form, }
-                    )
+class AdministratorLoginRequiredMixin(object):
+
+    @method_decorator(login_required)
```
Comment (tapasweni-pathak) [inline vms/organization/views.py]: @amruthasangeeth Hmm, is this taking care of 'admin is logged in or not' check?

File: vms/organization/views.py [hunk 4]
```diff
@@ -2,105 +2,80 @@
...
+            return render(request, 'vms/no_admin_rights.html')
         else:
-            form = OrganizationForm()
-            return render(
-                request,
-                'organization/create.html',
-                {'form': form, }
-                )
-    else:
-        return render(request, 'vms/no_admin_rights.html')
```
Comment (tapasweni-pathak) [inline vms/organization/views.py]: @amruthasangeeth Where are we handling this part in the new code?

File: vms/organization/views.py [hunk 5]
```diff
@@ -2,105 +2,80 @@
...
         else:
-            form = OrganizationForm(instance=organization)
-            return render(
-                request,
-                'organization/edit.html',
-                {'form': form, }
-                )
-    else:
-        return render(request, 'vms/no_admin_rights.html')
-
```
Comment (tapasweni-pathak) [inline vms/organization/views.py]: @amruthasangeeth This too?

File: vms/organization/views.py [hunk 6]
```diff
@@ -2,105 +2,80 @@
...
-
-@login_required
-def list(request):
-    organization_list = get_organizations_ordered_by_name()
```
Comment (tapasweni-pathak) [inline vms/organization/views.py]: @amruthasangeeth Here the list will be ordered by name, but this is not the case in our new code, isn't it?

File: vms/organization/models.py [hunk 7]
```diff
@@ -1,13 +1,14 @@
...
-            RegexValidator(
-                r'^[(A-Z)|(a-z)|(0-9)|(\s)|(\-)|(:)|(\')]+$',
-            ),
-        ],
-    )
+from django.core.validators import RegexValidator
+from django.db import models
+
+class Organization(models.Model):
+    id = models.AutoField(primary_key=True)
```
Comment (tapasweni-pathak) [inline vms/organization/models.py]: @amruthasangeeth Is there any other change in this file other than the newly added id?

File: vms/organization/views.py (hunk 3 above)
Comment (amruthasangeeth) [inline vms/organization/views.py]: @tapasweni-pathak Yes, It takes care of the same

File: vms/organization/views.py [hunk 8]
```diff
@@ -2,105 +2,80 @@
...
-
-
-@login_required
-def create(request):
-    if is_admin(request):
-        if request.method == 'POST':
-            form = OrganizationForm(request.POST)
-            if form.is_valid():
-                form.save()
-                return HttpResponseRedirect(reverse('organization:list'))
```
Comment (amruthasangeeth) [inline vms/organization/views.py]: @tapasweni-pathak This is handled by AdministratorLoginRequiredMixin.

File: vms/organization/views.py (hunk 4 above)
Comment (amruthasangeeth) [inline vms/organization/views.py]: @tapasweni-pathak Everything we have specified as template_name and form_class. FormView will take care of it.

File: vms/organization/views.py (hunk 6 above)
Comment (amruthasangeeth) [inline vms/organization/views.py]: @tapasweni-pathak Ya I will take care of that

File: vms/organization/views.py (hunk 2 above)
Comment (amruthasangeeth) [inline vms/organization/views.py]: @tapasweni-pathak It is handled by AdministratorLoginRequiredMixin

File: vms/organization/views.py (hunk 3 above)
Comment (tapasweni-pathak) [inline vms/organization/views.py]: @amruthasangeeth You above said that this is being taken care by the mixin?

File: vms/organization/views.py (hunk 4 above)
Comment (tapasweni-pathak) [inline vms/organization/views.py]: @amruthasangeeth Where is template_name and form_class are redirecting to no_admin_rights in such a case?

File: vms/organization/views.py (hunk 6 above)
Comment (tapasweni-pathak) [inline vms/organization/views.py]: @amruthasangeeth Please do this change and update the pull request or create a new one.

File: vms/organization/views.py (hunk 4 above)
Comment (amruthasangeeth) [inline vms/organization/views.py]: @tapasweni-pathak That is taken care by AdministratorLoginRequiredMixin

File: vms/organization/views.py (hunk 6 above)
Comment (amruthasangeeth) [inline vms/organization/views.py]: @tapasweni-pathak I have updated the PR.

File: vms/organization/views.py [hunk 9]
```diff
@@ -2,105 +2,80 @@
...
+        obj = Organization.objects.get(pk=org_id)
+        return obj
+
+
+class OrganizationListView(LoginRequiredMixin, ListView):
+    model_form = Organization
+    template_name = "organization/list.html"
+
+    def get_queryset(self):
+        organizations = Organization.objects.order_by('name')
```
Comment (amruthasangeeth) [inline vms/organization/views.py]: @tapasweni-pathak I have made the change here.

File: vms/organization/views.py (hunk 3 above)
Comment (amruthasangeeth) [inline vms/organization/views.py]: @tapasweni-pathak Yes it is taken care by AdministratorLoginRequiredMixin

File: vms/organization/templates/organization/delete.html (hunk 1 above)
Comment (amruthasangeeth) [inline vms/organization/templates/organization/delete.html]: @tapasweni-pathak DeleteView works just like that

**Pull request merged**